scribejava-core/src/main/java/com/github/scribejava/core/oauth/OAuth10aService.java

```diff
@@ -125,33 +193,46 @@ public void signRequest(final Token token, final AbstractRequest request) {
             request.addOAuthParameter(OAuthConstants.TOKEN, token.getToken());
         }
         config.log("setting token to: " + token);
-        addOAuthParams(request, token);
+        addOAuthParams(request, ((OAuth1Token) token).getSecret());
         appendSignature(request);
+
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public String getVersion() {
         return VERSION;
     }
 
-    /**
-     * {@inheritDoc}
-     */
     @Override
     public String getAuthorizationUrl(final Token requestToken) {
+        if (requestToken instanceof RequestToken) {
+            return getOAuth1AuthorizationUrl((RequestToken)requestToken);
+        } else  {
+            throw new IllegalArgumentException("requestToken must be an instance of RequestToken");
+        }
+    }
+
+    @Override
+    public String getOAuth1AuthorizationUrl(final RequestToken requestToken) {
         return api.getAuthorizationUrl(requestToken);
     }
 
-    private String getSignature(final AbstractRequest request, final Token token) {
+    /**
+     * OAuth 2 APIs are not implemented and will throw an UnsupportedOperationException.
+     * 
+     * @return throws exception
+     */
+    @Override
+    public String getOAuth2AuthorizationUrl() {
+        throw new UnsupportedOperationException("OAuth 2 authorization urls are not supported for OAuth 1 APIs");
+    }
```

Comment:
We have separated OAuth10aService and OAuth20Service. Users are getting specific version after invoking build(). 
I thinks there is no need to continue to support common OAuthService interface in case of methods like this with "throw new UnsupportedOperationException()" in one of the Implementation.